src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ArrayModelBinder.cs

```diff
@@ -16,16 +16,17 @@ namespace Microsoft.AspNetCore.Mvc.ModelBinding
     public class ArrayModelBinder<TElement> : CollectionModelBinder<TElement>
     {
         /// <inheritdoc />
-        public override Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
+        public override Task BindModelAsync(IModelBindingContext bindingContext)
         {
             if (bindingContext == null)
             {
                 throw new ArgumentNullException(nameof(bindingContext));
             }
+            Debug.Assert(bindingContext.Result == null);
 
             if (bindingContext.ModelMetadata.IsReadOnly)
             {
-                return ModelBindingResult.NoResultAsync;
+                return Internal.TaskCache.CompletedTask;
```

Comment:
It seemed to make more tactical to have a qualified reference to the single static than to expose all of the types in the internal namespace to a file that doesn't use anything else in there, no?